### PR TITLE
Add extra validation in e2e on instance via file api

### PR DESCRIPTION
**Pull request by annapendleton**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add extra validation in e2e on instance via file api
```

### CODE REVIEW

File: test/e2e/tests/single_zone_e2e_test.go [hunk 1]
```diff
@@ -54,9 +62,24 @@ var _ = Describe("Google Cloud Filestore CSI Driver", func() {
...
+			instanceUri := fmt.Sprintf(instanceURIFormat, proj, zone, volName)
+			_, err := fileInstancesService.Get(instanceUri).Do()
+			Expect(err).NotTo(BeNil(), "Could get deleted disk from cloud directly")
 		}()
 
-		// TODO validate the Filestore instance creation at the cloud provider layer
+		// Validate Disk Created
+		proj, zone, _ := instance.GetIdentity()
+		instanceUri := fmt.Sprintf(instanceURIFormat, proj, zone, volName)
+		inst, err := fileInstancesService.Get(instanceUri).Do()
```
Comment (saikat-royc) [inline test/e2e/tests/single_zone_e2e_test.go:75]: Let's create a helper function to get the instance.

File: test/remote/file.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	"time"
+
+	"github.com/golang/glog"
+	"golang.org/x/oauth2/google"
+	filev1beta1 "google.golang.org/api/file/v1beta1"
+)
+
+func GetFileClient() (*filev1beta1.Service, error) {
+	const retries = 10
+	const backoff = time.Second * 6
```
Comment (saikat-royc) [inline test/remote/file.go:31]: We can create global constants for this retry and backoff

File: test/e2e/tests/single_zone_e2e_test.go (hunk 1 above)
Comment (annapendleton) [reply test/e2e/tests/single_zone_e2e_test.go]: Done.

File: test/remote/file.go (hunk 2 above)
Comment (annapendleton) [reply test/remote/file.go]: Done.

**Pull request merged**